packages/app/src/runner/event-manager.ts

```diff
@@ -404,6 +408,9 @@ export class EventManager {
 
           if (!hideCommandLog) {
             this.reporterBus.emit('runnables:ready', runnables)
+            this.Mobx.runInAction(() => {
+              getMobxRunnerStore().setTotalTests(Cypress.runner.getTotalTestsBeforeFilter())
```

Comment:
In the long term, will the runner store's total Test count be used anywhere else outside of the header ui? If not, why not sent the total count with the runnables on `'runnables:ready'`? The reporter can track `filteredTestCount` and `totalTestCount`.

Reply:
This seems tough to answer at this point but @warrensplayer probably has the most insight into future iterations around this page.

Reply:
Refactored and grouped the run data on the root runnable to consolidate all of this logic. (Done in [5e431a0](https://github.com/cypress-io/cypress/pull/25265/commits/5e431a05532e2116e048ad3468ede9aa41ff6eda)!